### PR TITLE
Core(M): Describe scatter file templates.

### PR DESCRIPTION
**Pull request by GuentherMartin**

### CODE REVIEW

File: CMSIS/DoxyGen/Core/src/Template.txt [hunk 1]
```diff
@@ -377,6 +378,73 @@ The files for other compiler vendors differ slightly in the syntax, but not in t
 \verbinclude "Source/ARM/startup_Device_ac6.S"
 */
 
+/*=======0=========1=========2=========3=========4=========5=========6=========7=========8=========9=========0=========1====*/
+/**
+\page linker_sct_pg Scatter-Loading description file <device>_ac<5|6>.sct
+
+If C-Startup is used then it is necessary that also a scatter file for linking is used.
```
Comment (KeilChris) [inline CMSIS/DoxyGen/Core/src/Template.txt:385]: A scatter file for linking is required when using a \ref startup_c_pg "C startup file".

File: CMSIS/DoxyGen/Core/src/Template.txt [hunk 2]
```diff
@@ -377,6 +378,73 @@ The files for other compiler vendors differ slightly in the syntax, but not in t
...
+
+The \ref linker_sct_pg contains regions for:
+ - Code (Read only data, Execute only data)
```
Comment (KeilChris) [inline CMSIS/DoxyGen/Core/src/Template.txt:388]: - Code (read-only data, execute-only data)

File: CMSIS/DoxyGen/Core/src/Template.txt [hunk 3]
```diff
@@ -377,6 +378,73 @@ The files for other compiler vendors differ slightly in the syntax, but not in t
...
+ - RAM (Read/Write data, Zero initialized data)
```
Comment (KeilChris) [inline CMSIS/DoxyGen/Core/src/Template.txt:389]: - RAM (read/write data, zero-initialized data)

File: CMSIS/DoxyGen/Core/src/Template.txt [hunk 4]
```diff
@@ -377,6 +378,73 @@ The files for other compiler vendors differ slightly in the syntax, but not in t
...
+ - Stack
+ - Heap
+ - Stack Seal (Armv8-M/v8.1-M)
```
Comment (KeilChris) [inline CMSIS/DoxyGen/Core/src/Template.txt:392]: - Stack seal (Armv8-M/v8.1-M)

File: CMSIS/DoxyGen/Core/src/Template.txt [hunk 5]
```diff
@@ -377,6 +378,73 @@ The files for other compiler vendors differ slightly in the syntax, but not in t
...
+ - CMSE veneer (Armv8-M/v8.1-M)
+
+Within the scatter file the user needs to specify a set of macros
```
Comment (KeilChris) [inline CMSIS/DoxyGen/Core/src/Template.txt:395]: Within the scatter file, the user needs to specify a set of macros. The scatter file is passed through the C preprocessor which uses these macros to calculate the start address and the size of the different regions.

File: CMSIS/DoxyGen/Core/src/Template.txt [hunk 6]
```diff
@@ -377,6 +378,73 @@ The files for other compiler vendors differ slightly in the syntax, but not in t
...
+
+/*--------------------- CMSE Venner Configuration ---------------------------
+; <h> CMSE Venner Configuration
+;   <o0>  CMSE Venner Size (in Bytes) <0x0-0xFFFFFFFF:32>
+; </h>
+ *----------------------------------------------------------------------------*/
+#define __CMSEVENEER_SIZE    0x200
+\endcode
+
+The scatter file is passed through the C preprocessor and these macros are used to
```
Comment (KeilChris) [inline CMSIS/DoxyGen/Core/src/Template.txt:433]: Delete

File: CMSIS/DoxyGen/Core/src/Template.txt [hunk 7]
```diff
@@ -377,6 +378,73 @@ The files for other compiler vendors differ slightly in the syntax, but not in t
...
+calculate the start address and the size of the different regions.
```
Comment (KeilChris) [inline CMSIS/DoxyGen/Core/src/Template.txt:434]: Delete

**Review by KeilChris — CHANGES_REQUESTED** [review]
Needs some work; only minor changes.

File: CMSIS/DoxyGen/Core/src/Template.txt (hunk 1 above)
Comment (GuentherMartin) [reply CMSIS/DoxyGen/Core/src/Template.txt:385]: changed.

File: CMSIS/DoxyGen/Core/src/Template.txt (hunk 2 above)
Comment (GuentherMartin) [reply CMSIS/DoxyGen/Core/src/Template.txt:388]: changed

File: CMSIS/DoxyGen/Core/src/Template.txt (hunk 3 above)
Comment (GuentherMartin) [reply CMSIS/DoxyGen/Core/src/Template.txt:389]: changed

File: CMSIS/DoxyGen/Core/src/Template.txt (hunk 4 above)
Comment (GuentherMartin) [reply CMSIS/DoxyGen/Core/src/Template.txt:392]: changed

File: CMSIS/DoxyGen/Core/src/Template.txt (hunk 5 above)
Comment (GuentherMartin) [reply CMSIS/DoxyGen/Core/src/Template.txt:395]: changed

File: CMSIS/DoxyGen/Core/src/Template.txt (hunk 6 above)
Comment (GuentherMartin) [reply CMSIS/DoxyGen/Core/src/Template.txt:433]: deleted

File: CMSIS/DoxyGen/Core/src/Template.txt (hunk 7 above)
Comment (GuentherMartin) [reply CMSIS/DoxyGen/Core/src/Template.txt:434]: deleted

**Comment by JonatanAntoni** [comment]
@KeilChris, can you re-review the changes and resolve all conversations, please?

**Review by KeilChris — APPROVED** [review]

**Review by KeilChris — APPROVED** [review]

**Pull request merged**